wavecrestweb/kenworthy-machine
# #KM-20/ Set up TinaCMS

## Description
This PR sets up the configuration of TinaCMS, including schema to model the data that the CMS will hold, a visual editor to edit content, and queries to import the content to the app. The header and footer are not yet connected to the CMS.


## Related Tickets & Documents
Closes #KM-20


## Acceptance Criteria

- [x] a CMS has been researched and selected for ease of use for developers and robust visual editing for clients
- [x] the CMS data modeling matches the website content
- [x] the app queries the CMS to get real-time content




## What type of PR is this? (check all applicable)


|     | Type                       |
| --- | -------------------------- |
|  ✓ | :sparkles: New feature     |
|     | 🎨 Style                   |
|  ✓ | :hammer: Refactoring       |
|     | 🔥 Performance Improvements|
|     | :bug: Bug fix              |
|     | :100: Add tests            |
|  ✓ | :link: Update dependencies |
|     | :scroll: Docs              |


## Mobile & Desktop Screenshots/Recordings
![Screenshot 2024-02-02 at 3 52 22 PM](https://github.com/wavecrestweb/kenworthy-machine/assets/115492619/7259b9c2-6f6b-49e4-a00a-8d90a70279d5)



## What I learned
TinaCMS has real-time editing capabilities and what's often called a 'WYSIWYG' (what you see is what you get) interface. In development we'll run TinaCMS locally, but in production we'll use the self-hosted option of Tina Cloud.

## Testing steps

1. `git switch KM-20/feature-CMS`
2. `npm i` - so TinaCMS is installed!
3. `npm run dev` after installing
4. navigate to localhost:3000/admin (be sure to include the slug or the page will only load the header and footer) NOTE: it's a minute for the page to load, and another beat for the components to load
5. Once everything is loaded, you should see the visual editor on the left. It should be titled "Pages" and subtitled "blocks". Each block represents a separate component on the home page. Click into each block and play around with the text, you should see it change in real time on the browser screen on the right. Simply hit 'Reset' to undo any changes. Changes are really easy to make, so simply change something back if you accidentally saved, or want to see how long saving takes (it's fast!)
6. Feel free to check out the code to look into the setup, it'll be easiest to explain if I walk you through it!


## Added to documentation?

|     | Type                       |
| --- | -------------------------- |
|     | 📜 README.md    |
| ✓  | 🙅 no documentation needed    |


## What gif best describes this PR?
  ![woohoo](https://media.giphy.com/media/v1.Y2lkPTc5MGI3NjExaW1qMmxzeGZjNGZ4MmZ4Nm83cWl6Ym02YzAzanR4ZDBoejgxa3o1aSZlcD12MV9pbnRlcm5hbF9naWZfYnlfaWQmY3Q9Zw/vFnX2pErVfSGQGanTS/giphy.gif)
